contracts/contracts/strategies/aerodrome/AerodromeAMOStrategy.sol

```diff
@@ -0,0 +1,952 @@
+     * as 18 decimal point
+     *
+     * e.g. 0.2e18 means 20% WETH 80% OETHb
+     */
+    uint256 public poolWethShare;
+    /**
+     * @notice Specifies how the target WETH share of the pool defined by the `poolWethShare` can
+     * vary from the configured value after rebalancing. Expressed as 18 decimal point
+     */
+    uint256 public poolWethShareVarianceAllowed;
+    /**
+     * Share of liquidity to remove on rebalance expressed in 18 decimal points
+     */
+    uint128 public withdrawLiquidityShare;
+    /// @dev reserved for inheritance
+    int256[45] private __reserved;
+
+    /***************************************
+          Constants, structs and events
+    ****************************************/
+
+    /// @notice The address of the Wrapped ETH (WETH) token contract
+    address public immutable WETH;
+    /// @notice The address of the OETHb token contract
+    address public immutable OETHb;
+    /// @notice lower tick set to -1 representing the price of 1.0001 of WETH for 1 OETHb.
+    int24 public immutable lowerTick;
+    /// @notice lower tick set to 0 representing the price of 1.0000 of WETH for 1 OETHb.
+    int24 public immutable upperTick;
+    /// @notice tick spacing of the pool (set to 1)
+    int24 public immutable tickSpacing;
+    /// @notice the swapRouter for performing swaps
+    ISwapRouter public immutable swapRouter;
+    /// @notice the underlying AMO Slipstream pool
+    ICLPool public immutable clPool;
+    /// @notice the gauge for the corresponding Slipstream pool (clPool)
+    /// @dev can become an immutable once the gauge is created on the base main-net
+    ICLGauge public immutable clGauge;
+    /// @notice the Position manager contract that is used to manage the pool's position
+    INonfungiblePositionManager public immutable positionManager;
+    /// @notice helper contract for liquidity and ticker math
+    ISugarHelper public immutable helper;
+    /// @notice sqrtRatioX96Tick0
+    /// @dev tick 0 has value -1 and represents the lowest price of WETH priced in OETHb. Meaning the pool
+    /// offers less than 1 OETHb for 1 WETH. In other terms to get 1 OETHB the swap needs to offer 1.0001 WETH
+    /// this is where purchasing OETHb with WETH within the liquidity position is most expensive
+    uint160 public immutable sqrtRatioX96Tick0;
+    /// @notice sqrtRatioX96Tick1
+    /// @dev tick 1 has value 0 and represents 1:1 price parity of WETH to OETHb
+    uint160 public immutable sqrtRatioX96Tick1;
+    /// @dev tick closest to 1:1 price parity
+    ///      Correctly assesing which tick is closer to 1:1 price parity is important since it affects
+    ///      the way we calculate the underlying assets in check Balance
+    uint160 public immutable sqrtRatioX96TickClosestToParity;
+
+    error NotEnoughWethForSwap(uint256 wethBalance, uint256 requiredWeth); // 0x989e5ca8
+    error NotEnoughWethLiquidity(uint256 wethBalance, uint256 requiredWeth);
```

Comment:
nit: The other two errors have their signatures in the comments but not this one